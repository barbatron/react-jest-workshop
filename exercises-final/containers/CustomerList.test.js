import React from 'react'
import {shallow, mount} from 'enzyme'
import getStoreStub from '../store/Customers.stub'
import CustomerList from './CustomerList'

test('should render no customers', () => {
  const wrapper = shallow(<CustomerList store={getStoreStub().store} />)
  expect(wrapper.getElement()).toMatchSnapshot()
})

test('should render customers', () => {
  const {store} = getStoreStub([{name: 'Bob'}, {name: 'Joanna'}])
  const wrapper = shallow(<CustomerList store={store} />)
  expect(wrapper.getElement()).toMatchSnapshot()
})

test('should respond to store updates', () => {
  const {store, updateCustomers} = getStoreStub()
  const wrapper = shallow(<CustomerList store={store} />)
  expect(wrapper.getElement()).toMatchSnapshot()
  updateCustomers([{name: 'Jill'}, {name: 'Fred'}])
  wrapper.update();
  expect(wrapper.getElement()).toMatchSnapshot()
})

test('unsubscribe when unmounted', () => {
  const {unsubscribe, store} = getStoreStub()
  const wrapper = shallow(<CustomerList store={store} />)
  wrapper.unmount()
  expect(unsubscribe).toHaveBeenCalledTimes(1)
})
