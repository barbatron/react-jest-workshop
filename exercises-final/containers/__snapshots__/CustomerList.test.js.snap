// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render customers 1`] = `
<ListOfCustomers
  customers={
    Array [
      Object {
        "name": "Bob",
      },
      Object {
        "name": "Joanna",
      },
    ]
  }
/>
`;

exports[`should render no customers 1`] = `<NoCustomers />`;

exports[`should respond to store updates 1`] = `<NoCustomers />`;

exports[`should respond to store updates 2`] = `
<ListOfCustomers
  customers={
    Array [
      Object {
        "name": "Jill",
      },
      Object {
        "name": "Fred",
      },
    ]
  }
/>
`;
